base_db_api.py

```diff
@@ -0,0 +1,6 @@
+import abc
+
+class BaseDBAPI(abc.ABC):
+    @abc.abstractmethod
+    def is_site_safe(self, domain: str) -> bool:
+        raise NotImplementedError() 
```

Comment:
I'm not sure this would apply to all our DBs. maybe have something that returns an emun of the status we want
sus, malicious, safe